tcms/testcases/forms.py

```diff
@@ -622,7 +622,9 @@ class CaseTagForm(forms.Form):
 
     def populate(self, case_ids=None):
         if case_ids:
-            self.fields['o_tag'].queryset = TestTag.objects.filter(
-                testcase__in=case_ids).order_by('name').distinct()
+            tag_ids = TestCaseTag.objects.filter(
+                case__in=case_ids
+            ).values_list('tag').distinct()
+            self.fields['o_tag'].queryset = TestTag.objects.filter(pk__in=tag_ids).order_by('name')
```

Comment:
OK, this is counter intuitive but works. Django is able to follow backwards relationships filter. In this case the relationship is TestCaseTag -> TestTag but Django is able to figure out how to go backwards by using inner joins. 

IMO this is a really bad side of Django ORM because it may lead to unpredicted results. E.g. if you forget .distinct() at the end the results are different. Anyway I've added a comment in the code for future reference.